### ISSUE TITLE
mmc get phase invalid clockrate 

### ISSUE THREAD

**Issue by SolidHal** [SolidHal/PrawnOS#26]
Early in the boot kernel logs, these get thrown. 

```
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate
[    0.000000] rockchip_mmc_get_phase: invalid clk rate

```

I vaguely recall reading on the chrome os kernel somewhere that it is due to certain emmcs not liking the way speed tuning is handled, possibly in the commit logs for https://chromium.googlesource.com/chromiumos/third_party/kernel/+/chromeos-3.14/drivers/clk/rockchip/clk-mmc-phase.c

EDIT: non of the current untested patches fix this

**Comment by SolidHal** [comment]
The reason we don't see thin in the chromeos kernel is that the error isn't present in 
`chromeos-3.14/drivers/clk/rockchip/clk-mmc-phase.c`

The error has a pretty good description as to why it was added:
```
	/*
	 * The below calculation is based on the output clock from
	 * MMC host to the card, which expects the phase clock inherits
	 * the clock rate from its parent, namely the output clock
	 * provider of MMC host. However, things may go wrong if
	 * (1) It is orphan.
	 * (2) It is assigned to the wrong parent.
	 *
	 * This check help debug the case (1), which seems to be the
	 * most likely problem we often face and which makes it difficult
	 * for people to debug unstable mmc tuning results.
	 */
	if (!rate) {
		pr_err("%s: invalid clk rate\n", __func__);
		return -EINVAL;
	}
```

TODO: Try adding the two rate checks to the chromeos kernel

**Comment by SolidHal** [comment]
Reopening issue as these issues may be imapacting the other emmc related problems I'm. Submitted bug report.

**Comment by SolidHal** [comment]
according to some rockchip clock people this is likely a red herring and not indicative of a larger issue

**Comment by SolidHal** [comment]
this has been fixed in upstream, and will be included in 5.3. Backporting patches to 4.17.19, and eventually 4.19
